src/PseudoTypes/ValueOf.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/*
+ * This file is part of phpDocumentor.
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ *  @link      http://phpdoc.org
+ *
+ */
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Reflection\PseudoTypes;
+
+use phpDocumentor\Reflection\PseudoType;
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\Types\Mixed_;
+
+/**
+ * Value Object representing the `value-of` type.
+ *
+ * @psalm-immutable
+ */
+final class ValueOf implements PseudoType
```

Comment:
```suggestion
final class ValueOf extends Mixed_ implements PseudoType
```

As this class is a pseudo type of an Mixed_, users of this library may expect it to be an Mixed_.